src/webview/components/memory-table.tsx

```diff
@@ -220,33 +272,76 @@ export class MemoryTable extends React.PureComponent<MemoryTableProps, MemoryTab
             tableStyle: { minWidth: '30rem' },
             value: rows
         };
+    }
+
+    protected createScrollingBehaviorSpecificProperties(props: DataTableProps<MemoryRowData[]>): DataTableProps<MemoryRowData[]> {
+        if (this.props.scrollingBehavior === 'Auto-Append') {
+            return {
+                ...props,
+                pt: {
+                    wrapper: {
+                        onScroll: event => {
+                            if (!this.isLoading) {
+                                // Append new data only if we reach the bottom
+                                const element = event.target as HTMLDivElement;
+                                const distanceBottom = Math.abs(element.scrollHeight - element.scrollTop - element.clientHeight);
+
+                                if (distanceBottom < 1) {
+                                    this.appendMoreMemory();
+                                }
+                            }
+                        }
+                    }
+                }
+            };
+        } else {
+            return {
+                ...props,
+                footer: this.renderFooter()
+            };
+        }
+    }
 
+    /**
+     * This method asserts that we have sufficient rows visible to enable vertical scrollbars
+     */
+    protected ensureSufficientVisibleRowsForScrollbar(): void {
+        const datatableValues = this.datatableRef.current?.props.value;
+        if (datatableValues && datatableValues.length < this.rowsToAppendCount) {
+            // We have too few rows, we need to load more data
+            this.appendMoreMemory();
+        }
+    }
+
+    protected appendMoreMemory(): void {
+        if (!this.isLoading && MemoryState.hasMemory(this.props)) {
+            const options = this.createMemoryRowListOptions(this.props.memory, this.props);
+            const newCount = this.props.memoryReadArguments.count + options.wordsPerRow * this.rowsToAppendCount;
+            this.props.fetchMemory({ count: newCount });
+        }
     }
 
     protected onSelectionChanged = (event: DataTableSelectionCellChangeEvent<MemoryRowData[]>) => {
         this.setState(prev => ({ ...prev, selection: event.value }));
     };
 
     protected renderHeader(): React.ReactNode | undefined {
-        const { offset, count, fetchMemory, scrollingBehavior } = this.props;
-
         let memorySelect: React.ReactNode | undefined;
         let loading: React.ReactNode | undefined;
 
-        if (this.isShowMoreEnabled) {
+        if (this.isShowMoreEnabled && MemoryState.hasMemory(this.props)) {
```

Comment:
These two checks are paired in a couple of places, but I'm not sure I understand the need for both: under what conditions could the first check succeed (i.e. we have memory and it has bytes) and the second fail (checks for memory & memory read arguments)?